seqr/views/utils/project_context_utils.py

```diff
@@ -165,42 +165,46 @@ def add_project_tag_types(projects_by_guid, add_counts=False):
             ),
             'variantFunctionalTagTypes': VariantFunctionalData.FUNCTIONAL_DATA_TAG_TYPES,
         })
-        if add_counts:
-            family_counts.update(_add_tag_type_counts(project_guid, project_json['variantTagTypes']))
 
-    return family_counts
 
+def add_project_tag_type_counts(project, response_json, project_json=None):
+    response_json['projectsByGuid'] = {project.guid: project_json or {}}
+    add_project_tag_types(response_json['projectsByGuid'], project=project)
 
-def _add_tag_type_counts(project_guid, project_variant_tags):
-    project_tags = VariantTag.objects.filter(saved_variants__family__project__guid=project_guid)
-    project_notes = VariantNote.objects.filter(saved_variants__family__project__guid=project_guid)
+    saved_variants = SavedVariant.objects.filter(family__project=project)
+    project_tags = VariantTag.objects.filter(saved_variants__in=saved_variants)
+    project_notes = VariantNote.saved_variants.through.objects.filter(savedvariant_id__in=saved_variants)
 
     family_tag_type_counts = defaultdict(dict)
+
     note_tag_type = {
         'variantTagTypeGuid': 'notes',
         'name': 'Has Notes',
         'category': 'Notes',
         'description': '',
         'color': 'grey',
         'order': 100,
-        'numTags': project_notes.aggregate(count=Count('saved_variants__guid', distinct=True))['count'],
+        'numTags': project_notes.values_list('savedvariant_id').distinct().count(),
     }
 
-    mme_counts_by_family = project_tags.filter(saved_variants__matchmakersubmissiongenes__isnull=False) \
-        .values('saved_variants__family__guid').annotate(count=Count('saved_variants__guid', distinct=True))
+    mme_counts_by_family = saved_variants.filter(matchmakersubmissiongenes__isnull=False) \
```

Comment:
since this was only querying and counting off the saved variant models and not using the tag models at all, much more efficient to use the saved variant models directly